### ISSUE TITLE
Spurious "Too many template argument lists compared to parameter lists" error

### ISSUE THREAD

**Issue by pitrou** [michaeljones/breathe#407]
With the following inheritance structure:
```cpp

template <typename Type>
class PrimitiveBuilder : public ArrayBuilder {
 public:
  /// \brief The C type adequate for appending and representing data
  using value_type = typename Type::c_type;
};

template <typename T>
class NumericBuilder : public PrimitiveBuilder<T> {
 public:
  using typename PrimitiveBuilder<T>::value_type;
};
```

and the following directives:
```rst

.. doxygenclass:: arrow::PrimitiveBuilder
   :members:
```

I get this error:
```
Warning, treated as error:
/home/antoine/arrow/docs/source/cpp/api/builder.rst:34:Too many template argument lists compared to parameter lists. Argument lists: 1, Parameter lists: 0, Extra empty parameters lists prepended: 1. Declaration:
	arrow::PrimitiveBuilder<Type>::value_type
```

The error disappears if I remove the docstring for `PrimitiveBuilder::value_type`.

**Comment by jakobandersen** [comment]
It seems like this could be same problem as in #356.

**Comment by pitrou** [comment]
Well I have `INLINE_INHERITED_MEMB  = NO` here.

**Comment by jakobandersen** [comment]
Right, but I think the underlying mechanism in Breathe is the same: insert a declaration from a base class. It would be great if you could create a small example project with this example, to make further investigation easier.

**Comment by pitrou** [comment]
Ok, I've made a standalone reproducer project in https://github.com/pitrou/breathe-issue407

**Comment by emmenlau** [comment]
I'm also affected by this issue. Are there any known workarounds?

**Comment by vermeeren** [comment]
Should be fixed with #512, released in Breathe v4.17.0. Note that you also need Sphinx 3.x for recent Breathe versions.
